### ISSUE TITLE
Morris - Ashley and Bedding fill rules

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2148]
_Imported from trac ticket 2148.
Created by **sjohnson**
Opened in trac: Feb 14, 2013, 10:03:26 AM
Last modified in trac: May 20, 2013, 12:40:15 PM_
***
_by sjohnson:_
Morris - Ashley and Bedding fill rules.  Need logic defined from Morris

***
* kbower attached [2148 Ashley and Bedding Fill rules.doc](https://trac.retailarchitects.com/trac/attachment/ticket/2148/2148 Ashley and Bedding Fill rules.doc) on Feb 21, 2013, 1:40:40 PM
* sjohnson attached [Morris Ashely Fill rules.jpeg](https://trac.retailarchitects.com/trac/attachment/ticket/2148/Morris Ashely Fill rules.jpeg) on May 10, 2013, 11:02:57 AM
* sjohnson attached [Morris Filling Rules Mark Up.jpeg](https://trac.retailarchitects.com/trac/attachment/ticket/2148/Morris Filling Rules Mark Up.jpeg) on May 10, 2013, 11:03:32 AM

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 21, 2013, 1:41:09 PM:_

See attachment from Mark Bracey

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Feb 21, 2013, 4:06:39 PM:_

1) Do not fill Vendor Code ASHL unless
    If itm srt cd = ATP,  Fill if the delivery date is less than 90 days, do not fill if over 90
    Fill anyway if order sort cd = RDL, RDV, DCB or DCV (these are redeliveries)
    Fill ASHL if its a Pickup within 30 days
    If the line is already attached to a PO, don't fill it
    Refigure if the PU/DEL date is changed

2) Bedding defined as vendor codes SE11, ST14, SEAA
    * This one is about setting delivery date
       If XAL zone do whatever you want.
       If non XAL zone and if there is balance on order do not allow to set the delivery date.

    * This one is about filling rule
        Never fill for XAL zone.
        Ignore D location filling requests for now (these are docks and shouldn't be allocatable) revisit later.
        Fill within the days defined on the chart, don't fill beyond.
        Day ordered      Delivery/Pickup date
        Monday           Friday
        Tuesday          Saturday
        Wednesday        Wednesday
        Thu              Thu
        Fri              Thu
        Sat              Thu
        Sun              Thu

** Don't Unfill any Outlet in any of the above cases
**

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 21, 2013, 4:10:23 PM:_

Sameer, please see the new OrderDetail rule hook method:

```python
    def should_not_fill(self):
        """
        Rule hook. Return string reason for not filling if should not fill.
        """
        return None
```

For line level decisions (instead of order level decisions) about filling, override and use this function in the rules.

**Comment by kentbower** [comment]
_**sshah** trac comment on Feb 25, 2013, 5:44:48 PM:_

(Changeset [[3760]](../commit/68291a19836cf763414f583bc0fb2228bf24e78c))
* Fixes #2148 Morris rule for Ashley fill and Bedding fill

**Comment by kentbower** [comment]
_**sshah** trac status change on Feb 25, 2013, 5:44:48 PM:_ **closed**

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 27, 2013, 1:35:21 AM:_

(Changeset [[3786]](../commit/6906a931c48c4415c0a6b366370fbd218d528076))
* re #2148, we missed part about allowing outlet to fill for morris

**Comment by kentbower** [comment]
_**sshah** trac comment on Feb 28, 2013, 6:16:55 PM:_

(Changeset [[3825]](../commit/f01dc61297734ad8e01b5519b723f60a7a331893))
* Fixes #2148 Morris rule, if bedding lines are unfilled do not allow to set delivery date based on the days

**Comment by kentbower** [comment]
_**sshah** trac comment on Feb 28, 2013, 6:25:38 PM:_

(Changeset [[3826]](../commit/d10bbcc9c7e6a4f8e7e06cdb1064a0da4b14c32e))
* Fixes #2148

**Comment by kentbower** [comment]
_**sshah** trac comment on Feb 28, 2013, 6:51:50 PM:_

(Changeset [[3828]](../commit/6733c0fddda6570773a4f0f97734e92099e6401d))
* Fixes #2148 Morris rule unfilled bedding lines are displayed in error message if they are not filled when setting delivery date

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 28, 2013, 7:54:14 PM:_

(Changeset [[3829]](../commit/614f80e76c5c2dc97f68aaec0b6c5118b9d5d214))
* Fixes #2148

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Apr 22, 2013, 5:00:21 PM:_

Scott,
Can we exclude the RAC customer code from the rules for Bedding? The customer code is RACINC5501.
Thanks,
Darlene
___
Darlene,
Are you referring to the filling rules?  So if the customer is RAC, do you want the bedding to fill or not fill?  Is there anything else?
Thanks,
Scott
___
I think the concern is more that is doesn’t need to be paid in full to set delivery.  I think the filling rules will still apply.
Darlene

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Apr 22, 2013, 5:00:21 PM:_ **reopened**

**Comment by kentbower** [comment]
_**sshah** trac comment on Apr 22, 2013, 5:02:01 PM:_

(Changeset [[4056]](../commit/e71f255df86562e0d141527c7ad7bd7001b6c533))
* Fixes [#2274](2274) Morris rule change to print sales comment on invoice instead of delivery comments.
 * Fixes #2148 Morris Bedding Rule needs to give error if not paid in full for non RAC customer

**Comment by kentbower** [comment]
_**sshah** trac status change on Apr 22, 2013, 5:02:01 PM:_ **closed**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on May 8, 2013, 11:29:20 AM:_

We need to change the bedding rule for pickups. 

We want to allow a pickup order with a balance due to reserve merchandise as long as the pickup date is set within 7 days.

 Mark B

**Comment by kentbower** [comment]
_**sjohnson** trac status change on May 8, 2013, 11:29:20 AM:_ **reopened**

**Comment by kentbower** [comment]
_**sshah** trac comment on May 9, 2013, 11:02:07 AM:_

Mark,
Currently the filling rules apply to Vendors SE11, ST14 and SEAA.
You have messages appearing below not allowing save when there is a balance due on a non XAL.  To confirm the change you would like:  Allow any Shipping to be chosen if there is a balance due and the pick up date is within 7 days.  Then fill the pick up if possible (unless XAL) or follow the chart for presenting the earliest possible days for scheduling.
Please Clarify if I have anything misunderstood.
Thanks,
Scott

**Comment by kentbower** [comment]
_**sshah** trac comment on May 9, 2013, 11:02:35 AM:_

Yes.  That sounds right.   The new fill rule would apply to pickups specifically.

**Comment by kentbower** [comment]
_**sshah** trac comment on May 9, 2013, 11:38:17 AM:_

(Changeset [[4116]](../commit/806b90a4d468bdf2fd13095036247de1122f34b7))
* Fixes #2148 Morris bedding rule to allow setting shipping zone and possibly filling if order has balance due and pickup date is within 7 days. #mor

**Comment by kentbower** [comment]
_**sshah** trac status change on May 9, 2013, 11:38:17 AM:_ **closed**

**Comment by kentbower** [comment]
_**sjohnson** trac status change on May 10, 2013, 11:00:57 AM:_ **reopened**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on May 10, 2013, 11:04:39 AM:_

Added marked up rules for documentation.

**Comment by kentbower** [comment]
_**sjohnson** trac status change on May 10, 2013, 11:04:39 AM:_ **closed**

**Comment by kentbower** [comment]
_**sshah** trac comment on May 20, 2013, 12:40:15 PM:_

(Changeset [[4155]](../commit/e92c8ad964548a83add0ccecae00319f5a29ea84))
* Fixes #2148 Morris bedding rule to possibly fill if order has balance due and pickup date is within 7 days. #mor
